File: app/src/main/java/cr/proyect/una/globales/info/domain/use_case/DeleteProductUseCase.kt
package cr.proyect.una.globales.info.domain.use_case

import cr.proyect.una.globales.info.domain.repository.ProductRepository
import javax.inject.Inject

class DeleteProductUseCase @Inject constructor(
    private val productRepository: ProductRepository
) {
    suspend operator fun invoke(productId: String) {
        productRepository.deleteProduct(productId)
    }
}

File: src/domain/useCases/deleteProductUseCase.ts
import { ProductRepository } from "../repositories/productRepository";

export class DeleteProductUseCase {
  private productRepository: ProductRepository;

  constructor(productRepository: ProductRepository) {
    this.productRepository = productRepository;
  }

  async execute(productId: number): Promise<boolean> {
    return await this.productRepository.deleteProduct(productId);
  }
}
